File: apps/agent-service/skills/requirements-outline/scripts/render.mjs
let input = '';
for await (const chunk of process.stdin) input += chunk;
const { args, outputDir } = JSON.parse(input);
if (typeof args.title !== 'string' || !Array.isArray(args.sections)) throw new Error('title must be a string; sections must be an array');
const lines = [`# ${args.title.replaceAll('\n', ' ')}`, ''];
for (const section of args.sections) {
  if (typeof section.heading !== 'string' || !Array.isArray(section.items)) throw new Error('each section needs heading and items');
  lines.push(`## ${section.heading.replaceAll('\n', ' ')}`, '');
  for (const item of section.items) {
    if (typeof item !== 'string') throw new Error('items must contain strings');
    lines.push(`- ${item.replaceAll('\n', '\n  ')}`);
  }
  lines.push('');
}
const { writeFile } = await import('node:fs/promises');
const { join } = await import('node:path');
await writeFile(join(outputDir, 'requirements.md'), lines.join('\n'), 'utf8');
console.log(JSON.stringify({ generated: 'requirements.md' }));

File: apps/agent-service/skills/requirements-outline/scripts/render.py
# -*- coding: utf-8 -*-
"""Python 3.7+; standard library only. Input envelope is JSON on stdin."""
import json
import os
import sys

request = json.load(sys.stdin)
args = request['args']
if not isinstance(args.get('title'), str) or not isinstance(args.get('sections'), list):
    raise ValueError('title must be a string; sections must be an array')
lines = ['# ' + args['title'].replace('\n', ' '), '']
for section in args['sections']:
    if not isinstance(section.get('heading'), str) or not isinstance(section.get('items'), list):
        raise ValueError('each section needs heading and items')
    lines.extend(['## ' + section['heading'].replace('\n', ' '), ''])
    for item in section['items']:
        if not isinstance(item, str):
            raise ValueError('items must contain strings')
        lines.append('- ' + item.replace('\n', '\n  '))
    lines.append('')
with open(os.path.join(request['outputDir'], 'requirements.md'), 'w', encoding='utf-8') as output:
    output.write('\n'.join(lines))
print(json.dumps({'generated': 'requirements.md'}, ensure_ascii=False))
